src/ILLink.Shared/DiagnosticId.cs

```diff
@@ -223,5 +225,24 @@ public static class DiagnosticIdExtensions
 				>= 3054 and <= 3055 => MessageSubCategory.AotAnalysis,
 				_ => MessageSubCategory.None,
 			};
+
+		public static string GetDiagnosticCategory (this DiagnosticId diagnosticId)
```

Comment:
nit: this can be reduced to:

```c#
		public static string GetDiagnosticCategory (this DiagnosticId diagnosticId) => (int)diagnosticId switch
        {
            > 2000 and < 3000 => DiagnosticCategory.Trimming,
            >= 3000 and < 3050  => DiagnosticCategory.SingleFile,
            >= 3050 and <= 6000 => DiagnosticCategory.AOT,
            _ => throw new ArgumentException ($"The provided diagnostic id '{diagnosticId}' does not fall into the range of supported warning codes 2001 to 6000 (inclusive).")
        };
```